litestar/openapi/config.py

```diff
@@ -133,7 +134,7 @@ def to_openapi_schema(self) -> OpenAPI:
         return OpenAPI(
             external_docs=self.external_docs,
             security=self.security,
-            components=cast("Components", self.components),
+            components=deepcopy(self.components),  # deepcopy prevents mutation of the config's components
```

Comment:
The main reason we need this is that `DEFAULT_OPENAPI_CONFIG` is a global, and if `components.schemas` is defined on the config object, then it is otherwise the same dict that makes its way onto the openapi schema. That dict is then mutated as the schema is built, which is actually mutating the components value on the global config.

In the implementation that is superseded by this PR, this was not required b/c the schemas dict on the openapi object would be overwritten every time an application is constructed, if `OpenAPIConfig.components`, or `OpenAPIConfig.components.schemas` were not defined.

Reply:
> The main reason we need this is that `DEFAULT_OPENAPI_CONFIG` is a global

I'd really like to get rid of that global, but I think that would technically breaking. Is there any reason besides this to keep it around?

Reply:
> > The main reason we need this is that `DEFAULT_OPENAPI_CONFIG` is a global
> 
> I'd really like to get rid of that global, but I think that would technically breaking. Is there any reason besides this to keep it around?

Then the default would have to be some special marker instance and then we'll have to create the default config after checking if the value is the marker. I don't think this is breaking since the default config will not change and the user can still set `None` to disable OpenAPI.